### DISCUSSION TITLE
Provide a Docker image for ent/contrib development/CI.

### DISCUSSION THREAD

**Discussion by tarrencev** [ent/ent#1676] [Ideas]
It would be useful to provide a canonical ent runtime for reproducing ci locally. Right now, it can be tricky to match generated outputs due to dependencies on the environment. We could provide an `ent` runtime docker file which contains all dependencies necessary to run/build ent, migrate the ci workflows to using it as the base image, and provide an easy way for contributors to reproduce ci outputs locally.

**Comment by rotemtam** [comment]
Just to be more specific (completing details I've discussed with you and @a8m over slack) - the goal is to make the dev/CI easily reproducible locally such that `go generate ./...` output is consistent for example. Things like `protoc` version etc.

Currently, the CI environment is described in  [.github/workflows/ci.yml](https://github.com/ent/contrib/blob/master/.github/workflows/ci.yml). Moving it to be docker-based will be much easier to work with locally.

**Comment by rotemtam** [comment]
For anyone hitting this, until we properly resolve this, here's a Docker image that should work the same as the `ent/contrib` GitHub Action environment:

```dockerfile
FROM golang:1.16.3
RUN apt update && apt install unzip
RUN curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v3.17.3/protoc-3.17.3-linux-x86_64.zip && \
 unzip -o protoc-3.17.3-linux-x86_64.zip -d ./proto && \
 chmod 755 -R ./proto/bin && \
 cp ./proto/bin/protoc /usr/bin/ && \
 cp -R ./proto/include/* /usr/include/
RUN  go get google.golang.org/protobuf/cmd/protoc-gen-go &&
    go get google.golang.org/grpc/cmd/protoc-gen-go-grpc
```
